srcpkgs/hplip-plugin/template

```diff
@@ -0,0 +1,61 @@
+# Template file for 'hplip-plugin'
+# When possible, keep this in sync with hplip-plugin.
+pkgname=hplip-plugin
+version=3.22.10
+revision=1
+archs="i686 x86_64 armv[67]l aarch64"
+hostmakedepends="tar"
+short_desc="Binary plugin for HPs hplip printer driver library"
+maintainer="biopsin <biopsin@teknik.io>"
+license="custom:proprietary"
+homepage="https://developers.hp.com/hp-linux-imaging-and-printing/binary_plugin.html"
+changelog="https://developers.hp.com/hp-linux-imaging-and-printing/release_notes"
+distfiles="https://developers.hp.com/sites/default/files/hplip-${version}-plugin.run"
+checksum=bb9648ea0626a9b3cfa29e260348d5d617f328c55b66eda346384c04f4740b1f
+repository=nonfree
+restricted=yes
+
+ignore_elf_dirs="/usr/share/hplip"
+
+do_extract() {
+	/bin/sh ${XBPS_SRCDISTDIR}/${pkgname}-${version}/hplip-${version}-plugin.run \
+	--noexec --target ${wrksrc}
+}
+
+do_install() {
+	vmkdir usr/share/hplip/data/firmware
+	vmkdir usr/share/hplip/fax/plugins
+	vmkdir usr/share/hplip/prnt/plugins
+	vmkdir usr/share/hplip/scan/plugins
+	vmkdir usr/share/licenses/hplip-plugin
+	vmkdir var/lib/hp
+
+	case "$XBPS_TARGET_MACHINE" in
+		i686) _arch='x86_32' ;;
+		x86_64) _arch='x86_64' ;;
+		armv[67]l) _arch='arm32' ;;
+		aarch64) _arch='arm64' ;;
+	esac
+
+	vcopy plugin.spec usr/share/hplip/
+	vcopy hp_laserjet_*.fw.gz usr/share/hplip/data/firmware/
+	vcopy fax_marvell-"$_arch".so usr/share/hplip/fax/plugins/
+	vcopy hbpl1-"$_arch".so usr/share/hplip/prnt/plugins/
+	vcopy lj-"$_arch".so usr/share/hplip/prnt/plugins/
+	vcopy bb_*-"$_arch".so usr/share/hplip/scan/plugins/
+	vlicense license.txt
+
+	cat << EOF > hplip.state
+[plugin]
+installed = 1
+eula = 1
+version = ${version}
+EOF
+	vcopy hplip.state var/lib/hp
+
+	find "${DESTDIR}/usr/share/hplip" -type f -name "*.so" | while read f; do
+		lib_dir="${f%/*}"
+		lib_name="${f##*/}"
+		ln -sf "$lib_name" "$lib_dir/${lib_name%%-*}.so"
+	done
```

Comment:
Use `vinstall`, that's what it was made for. Not only does it make directories as needed, but we can remove the group-write bit from all of these files.

Also, I don't see why we need the original `-${_arch}` in the names if you're just going to symlink these to generic names (presumably `hplip` expects to find generic names anyway), so just drop the symlink and install the files with their generic names in the first place. (If that doesn't work, just drop the fourth argument from `vinstall` and keep your `find` loop.)

```suggestion
	local _arch _f

	case "$XBPS_TARGET_MACHINE" in
		i686) _arch='x86_32' ;;
		x86_64) _arch='x86_64' ;;
		armv[67]l) _arch='arm32' ;;
		aarch64) _arch='arm64' ;;
	esac

	vinstall plugin.spec 0644 usr/share/hplip

	for _f in hp_laserjet_*.fw.gz; do
		vinstall "${_f}" 0644 usr/share/hplip/data/firmware
	done

	vinstall "fax_marvell-${_arch}.so" 0755 \
		usr/share/hplip/fax/plugins fax_marvell.so

	vinstall "hbpl1-${_arch}.so" 0755 \
		usr/share/hplip/prnt/plugins hbpl1.so

	vinstall "lj-${_arch}.so" 0755 \
		usr/share/hplip/prnt/plugins lj.so

	for _f in bb_*-"${_arch}.so"; do
		vinstall "${_f}" 0755 \
			usr/share/hplip/scan/plugins "${_f/-${_arch}}"
	done

	vlicense license.txt

	cat <<-EOF > hplip.state
		[plugin]
		installed = 1
		eula = 1
		version = ${version}
		EOF

	vinstall hplip.state 0644 var/lib/hp
```